### PR TITLE
test: fixes for multiplatform ChakraCore CI

### PR DESCRIPTION
**Pull request by joaocgreis**
This PR should make the ChakraCore CI jobs pass on the current xplat head. This includes 3 commits:

- 7f8b1cefa69f6a010df478d2886250f0d064f9c8: Fixes some differences to `node/master` that probably resulted from merge operations. Most are whitespace differences.
- b803decba030eee9bf4275d75c942547234a0bd9: Mark all tests that fail in CI for ChakraCore but not for V8 as flaky in the status files. This makes the build yellow, not green, so that these tests are not forgotten. Also uses `common.skip()` coherently throughout the tests that are skipped for ChakraCore (edited, was in a separate commit previously).
- db752808343f070fb5db0bcb221683e346f01ec1: skips test-child-process-pass-fd, it makes Jenkins fail the test run but the test itself passes locally. I will investigate further and revert this commit as soon as possible.

Tests that have issues with ChakraCore should fall in one of two categories:
- Tests that are V8 specific: Should be skipped in the beginning of the test code with `common.skip()` (this allows the build to be green).
- Tests that are not _yet_ working with ChakraCore, but should in the future: should be marked as flaky in the status files (this makes the build yellow, so that these tests are not forgotten).

cc @nodejs/node-chakracore 

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-guidelines)

##### Affected core subsystem(s)

test

### CODE REVIEW

**Comment by joaocgreis** [comment]
CI: https://ci.nodejs.org/view/All/job/chakracore-test-pull-request/13/

**Comment by agarwal-sandeep** [comment]
LGTM

**Review by kfarnung — APPROVED** [review]
LGTM

**Review by kunalspathak — COMMENTED** [review]
I would first merge 47f8f74 in node-chakracore without manually deleting these files as a separate commit to avoid merge conflicts.

**Comment by joaocgreis** [comment]
@kunalspathak what do you mean? https://github.com/nodejs/node-chakracore/commit/47f8f7462fb198aa27ede602c43786bdbfda37a2 was already merged in the last merge PR (https://github.com/nodejs/node-chakracore/pull/211), so it's already an ancestor of this commit. Those files should have been deleted as part of the merge.

**Comment by kunalspathak** [comment]
I see, in that case are you saying that those files are still present in node-chakracore because of bad merge? Why is the need to delete them again in your commit?

**Comment by joaocgreis** [comment]
@kunalspathak yes, they were not deleted during the merge. I don't see a reason to keep them here, do they test functionality that is still relevant in node-chackracore but not v8? I compared the `test/` from `xplat` with the `node/master` parent of the last merge commit, and removed every change that I did not see a reason for (that I believe should have happened in the merges). It is useful to have the diff as clean as possible, to identify what changed in node-chakracore.

**Comment by kunalspathak** [comment]
> It is useful to have the diff as clean as possible, to identify what changed in node-chakracore.

@joaocgreis , I agree with you. I was just trying to understand if this was the result of bad merge. Thanks for confirming and yes, please delete those files since they are already deleted in nodejs/master.

Your changes LGTM.

**Comment by joaocgreis** [comment]
Rebased and updated: tests should now fall strictly in the two categories mentioned in the OP. @kunalspathak thanks for your help with this.

@agarwal-sandeep @kfarnung this is now different than the version you previously stamped, let me know if you want time to review this again. Thanks!

CI: https://ci.nodejs.org/job/chakracore-test-pull-request/23/

**Comment by joaocgreis** [comment]
I added a commit to fix a linter error. Can someone take a look to make sure it's good to land? Thanks!

File: test/sequential/sequential.status [hunk 1]
```diff
@@ -22,4 +22,5 @@ test-benchmark-child-process  : PASS,FLAKY
 [$jsEngine==chakracore]
 
 [$jsEngine==chakracore && $system==linux]
-test-child-process-pass-fd : PASS,FLAKY
+# This test crashes Jenkins in Linux CI
+test-child-process-pass-fd : SKIP
```
Comment (kunalspathak) [inline test/sequential/sequential.status]: if this test crashes Jenkins in Linux CI, why is it under `$jsEngine==chakracore` condition? Or is it only crashing CI setup for chakracore?

File: lib/trace_mgr.js [hunk 2]
```diff
@@ -353,7 +353,7 @@ function checkGlobalShouldEmit(emitKind, optInfo) {
     } else {
       return (emitOptions.emitOnExit === 'error') && (optInfo !== 0);
     }
-  } else if (emitKind == 'emitOnException' || emitKind == 'emitOnSigInt') {
+  } else if (emitKind === 'emitOnException' || emitKind === 'emitOnSigInt') {
```
Comment (kunalspathak) [inline lib/trace_mgr.js]: Kyle already fixed it in https://github.com/nodejs/node-chakracore/commit/4751de59621ce28f42a4a59f718308c681acaaff#diff-d3e8e0bdb7af6cb1b0059f4d2d23e4b1

**Review by kfarnung — APPROVED** [review]

File: lib/trace_mgr.js (hunk 2 above)
Comment (joaocgreis) [reply lib/trace_mgr.js]: Thanks! I'll remove the commit.

File: test/sequential/sequential.status (hunk 1 above)
Comment (joaocgreis) [reply test/sequential/sequential.status]: Only for chakracore. I'm investigating this, I'll report when I find the cause.

**Comment by joaocgreis** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/27/

Landed in https://github.com/nodejs/node-chakracore/compare/4751de59621ce28f42a4a59f718308c681acaaff...7e6de45b0804902459528206ff0c6050135cb2db

**Pull request merged**